src/index.js

```diff
@@ -137,14 +150,53 @@ const getPointsLeftByDay = async (sprint, start) => {
       pointsLeftByDay[i] = 0;
     }
   }
-  return pointsLeftByDay;
-};
 
-const generateChart = async (data, labels, filenamePrefix) => {
-  const pointsPerDay = data[0] / (labels[labels.length - 1] - 1);
-  const constantLine = labels.map(
-    (label) => data[0] - pointsPerDay * (label - 1)
+  const initialPoints = pointsLeftByDay[0];
+  /** range from 0 to 6 */
+  const startDayOfWeek = start.day();
+  /** range from 0 to 13 but always fulfils startDayOfWeek <= endDayOfWeek < startDayOfWeek + 7 */
+  const endDayOfWeek = end.day() >= startDayOfWeek ? end.day() : end.day() + 7;
+  // if startDayOfWeek == endDayOfWeek, then there is no extra "short week"
+  const extraSat = startDayOfWeek <= 6 && endDayOfWeek > 6;
+  const extraSun =
+    (startDayOfWeek <= 0 && endDayOfWeek > 0) ||
+    (startDayOfWeek <= 7 && endDayOfWeek > 7);
+  const numberOfWeekends =
+    // number of complete weeks
+    Math.floor(numDaysInSprint / 7) * 2 +
+    // weekends in the extra "short week"
+    extraSat +
+    extraSun;
+  const numberOfWeekdays = numDaysInSprint - numberOfWeekends;
+  const pointsPerDay =
+    initialPoints / (INCLUDE_WEEKENDS ? numDaysInSprint : numberOfWeekdays - 1);
+
+  log.info(
+    JSON.stringify({
+      initialPoints,
+      numDaysInSprint,
+      numberOfWeekends,
+      numberOfWeekdays,
+      pointsPerDay,
+    })
   );
+
+  const guideline = [];
+  const cur = moment(start);
+  for (let index = 0; index < numDaysInSprint; index++, cur.add(1, "days")) {
+    guideline[index] =
+      index === 0
+        ? initialPoints
+        : guideline[index - 1] -
+          (!INCLUDE_WEEKENDS && cur.day() < 2 ? 0 : pointsPerDay);
+    // deduct based on the previous day
+    // if cur.day() is 0 or 1 means the previous day was 6 (sat) or 0 (sun)
+  }
```

Comment:
Looks a bit complex 😅  Before we dive in to understand what this chunk is doing, let me share this information with you:
* We can actually choose to run the integration on only the days we need it to run. E.g. MON-FRI only
* That means the daily summary will exclude points per day info for weekends already
* How might we then generate the chart without these missing days in the x-axis in a simpler way?

Feel free to PM to clarify if that makes it faster to discuss.

Reply:
So, the idea is that the chart should not show weekends with flat lines at all? Just a straight line down across all workdays?

Reply:
I think the main complex part here is calculating the number of weekdays across the whole sprint, which is what the main chunk of the code 154-172 is doing. I'm not sure how we could use the integration to generate that though. Any ideas?